source/utilities.cc

```diff
@@ -104,6 +105,43 @@ namespace aspect
       return static_cast<bool>(ifile); // only in c++11 you can convert to bool directly
     }
 
+    void
+    load_and_distribute_file_content(const std::string &filename,
+                                     const MPI_Comm &comm,
+                                     std::stringstream &datastream)
+    {
+      // Reset datastream
+      datastream.str(std::string());
+
+      if (Utilities::MPI::this_mpi_process(comm) == 0)
+        {
+          std::ifstream filestream(filename);
+          AssertThrow (filestream,
+                       ExcMessage (std::string("Could not open file <") + filename + ">."));
+
+          // Read data from disk
+          filestream >> datastream.rdbuf();
```

Comment:
By the way, the description at the page says that this operator "Extracts as many characters as possible from the stream and inserts them into the output sequence controlled by the stream buffer object pointed by sb (if any), until either the input sequence is exhausted or the function fails to insert into the object pointed by sb."

I assume that means that there may be cases where the streambuf is simply not large enough to hold the content of the file. I don't know whether that can happen in practice, but maybe it's worth testing that the input file has been read in its entirety by checking something of the sort
```
  AssertThrow (!filestream.good(), ...);
```
or similar (maybe `filestream.eof()` or some other predicate).